lib/array_intersection.rb

```diff
@@ -1,3 +1,11 @@
 def intersection(list1, list2)
-  
+  hash = {}
+  intersections = []
+  list1.each { |element| hash[element] = true }
+
+  list2.each do |element|
```

Comment:
I like how you use whitespace to separate the the multi-line `each` statement from the single-line statements. You may also consider using a multi-line `each` for the `list1` iteration, it would make the code scan slightly faster for a reader because the two loops would structurally look different than the variable declarations.

Reply:
Thanks for the feedback bion! I'll add time and space complexity for these algorithms this evening